qpc/insights/upload.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2018 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public License,
+# version 3 (GPLv3). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv3
+# along with this software; if not, see
+# https://www.gnu.org/licenses/gpl-3.0.txt.
+#
+"""Upload is used to upload files through the insights client."""
+
+from __future__ import print_function
+
+import subprocess
+import sys
+
+import qpc.insights as insights
+from qpc import messages
+from qpc.clicommand import CliCommand
+from qpc.insights.utils import (check_insights_install,
+                                check_successful_upload,
+                                insights_command,
+                                test_insights_command)
+from qpc.request import GET
+from qpc.translation import _
+from qpc.utils import (write_file)
+
+# pylint:disable=no-member
+from requests import codes
+
+
+# pylint: disable=too-few-public-methods
+class InsightsUploadCommand(CliCommand):
+    """Defines the Insights command.
+
+    This command is for uploading QPC reports throught the insights client.
+    """
+
+    SUBCOMMAND = insights.SUBCOMMAND
+    ACTION = insights.UPLOAD
+
+    def __init__(self, subparsers):
+        """Create command."""
+        CliCommand.__init__(self, self.SUBCOMMAND, self.ACTION,
+                            subparsers.add_parser(self.ACTION), GET,
+                            insights.REPORT_URI, [codes.ok])
+        self.parser.add_argument('--report', dest='report_id',
```

Comment:
In our other commands relating to reports, especially downloading (qpc report details/deployments), we can retrieve the report by providing one of two options, either a report id like you do here, or a scan job id. I feel like we should add the option for uploading a report based on scan job to be consistent with our other commands. I don't think it's too much extra to add. What are your thoughts?

Reply:
That's probably a good idea.  I personally like report id, but consistency is important.  Let's add scan job id.